### PR TITLE
fix(content-sharing): Use marker-based paging for users and groups

### PR DESCRIPTION
**Pull request by sahiga**
Update users and groups retrieval in the ContentSharing Element to use marker-based paging instead of offset-based paging.

### CODE REVIEW

File: src/api/Groups.js [hunk 1]
```diff
@@ -72,33 +53,6 @@ class Groups extends Base {
...
-    ): void {
-        this.errorCode = ERROR_CODE_FETCH_ENTERPRISE_GROUPS;
-        this.get({
-            id,
-            successCallback,
-            errorCallback,
-            url: this.getGroupsInEnterpriseUrl(filterTerm),
-            requestData,
-        });
-    }
```
Comment (manishPh) [inline src/api/Groups.js:101]: Did groups and users APIs change? Can we still have offset based pagination for these endpoints?

Comment (sahiga) [reply src/api/Groups.js:101]: I added these functions, so they're not currently being used for anything else in BUIE. I can add them back in, though. Thoughts?

Comment (manishPh) [reply src/api/Groups.js:101]: @sahiga, if you are removing the same functions that you added recently, then it's ok.

File: src/api/MarkerBasedAPI.js [hunk 2]
```diff
@@ -62,6 +62,7 @@ class MarkerBasedApi extends Base {
                 ...requestData,
                 marker,
                 limit,
+                usemarker: true,
```
Comment (swfree) [inline src/api/MarkerBasedAPI.js:65]: just confirming - this change won't impact any other APIs that use this MarkerBasedAPI? i.e. if they don't have this query param implemented, it'll just be ignored?

**Review by swfree — DISMISSED** [review]

File: src/api/MarkerBasedAPI.js (hunk 2 above)
Comment (booc0mtaco) [reply src/api/MarkerBasedAPI.js:65]: in most cases, we use camelCase, but this one is all lowercase. Is that what we have to use here?

Comment (sahiga) [reply src/api/MarkerBasedAPI.js:65]: Good question. The `usemarker` parameter enforces the use of marker-based paging when both offset-based paging and marker-based paging are available. It shouldn't affect the other files that use this API, but to be safe, I'll move this parameter to the `MarkerBasedGroups` and `MarkerBasedUsers` files.

Comment (sahiga) [reply src/api/MarkerBasedAPI.js]: @booc0mtaco Yeah, this parameter is specified as all lowercase in the docs. I was surprised by the format, too.

**Pull request merged**